functional-samples/tutorial.custom-cursor/service-worker.js

```diff
@@ -0,0 +1,9 @@
+chrome.runtime.onMessage.addListener(async (msg) => {
+  if (msg === 'page-click') {
+    const data = await chrome.storage.local.get({ counter: 0 });
+    let counter = data['counter'];
+    counter++;
+    chrome.storage.local.set({ counter });
+    await chrome.action.setBadgeText({ text: counter.toString() });
```

Comment:
is the toString call needed? Don't care either way, id just be surprised if it was

Reply:
It is actually!

<img width="936" alt="Screenshot 2024-07-30 at 14 59 37" src="https://github.com/user-attachments/assets/13fa9c1c-d405-4b0a-8dc8-5987ad985b40">

Reply:
my loosey-goosey-everything-is-an-object-anyway-javascript background is deeply offended ![](https://gifrific.com/wp-content/uploads/2012/04/ryan-gosling-reaction.gif)